src/platform/Linux/bluez/ChipDeviceScanner.cpp

```diff
@@ -112,31 +112,30 @@ void ChipDeviceScanner::Shutdown()
         },
         this);
 
-    mIsInitialized = false;
+    mCurrentState = ChipDeviceScannerState::SCANNER_UNINITIALIZED;
 }
 
 CHIP_ERROR ChipDeviceScanner::StartScan(System::Clock::Timeout timeout)
 {
+    CHIP_ERROR err              = CHIP_NO_ERROR;
+    static bool startingScanner = false; // To prevent function reentrancy
+
     assertChipStackLockedByCurrentThread();
-    VerifyOrReturnError(mIsInitialized, CHIP_ERROR_INCORRECT_STATE);
-    VerifyOrReturnError(!mIsScanning, CHIP_ERROR_INCORRECT_STATE);
+    // TODO: The caller needs to handle the busy status
+    VerifyOrReturnError(!startingScanner, CHIP_ERROR_BUSY);
```

Comment:
Instead of that, how about using a mutex to prevent race when calling this function from different threads? Then the caller will not have to deal with the BUSY state. The same thing for stop scan.